src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy

```diff
@@ -117,6 +119,26 @@ class GrabbitConfiguration {
         )
     }
 
+    /**
+     * Removes '/.' from {@code pathConfigurations}'s path if exists in the end, and Sorts the {@code pathConfigurations}
+     * on the basis of paths' depth
+     *
+     * @param pathConfigurations
+     *          Collection of {@link PathConfiguration} to sort
+     * @return
+     *          Sorted Collection of {@link PathConfiguration}
+     */
+    private static Collection<PathConfiguration> getSortedPathConfigurationsByDepth(
+            @Nonnull final Collection<PathConfiguration> pathConfigurations) {
+
+        return pathConfigurations.sort(false) { PathConfiguration it ->
+            final String thePath = it.path
+            //'/.' means 'this node', so we don't count it as an additional directory level
+            it.path = thePath[-2..-1] == "/." ? thePath - "/." : thePath
```

Comment:
by using, "it.path" we are modifying the original input argument's members. That is why I created "thePath" - so we can mutate it if we need to. However, I did make a mistake by making it final. Habit. 🐹 

```groovy
private static List<PathConfiguration> getSortedPathConfigurationsByDepth(@Nonnull final Collection<PathConfiguration> pathConfigurations) {
  return pathConfigurations.sort(false) { 
    String thePath = it.path
    thePath = thePath[-2..-1] == "/." ? thePath - "/." : thePath
    return thePath.count('/')
  }
}
```